### DISCUSSION TITLE
How to Access $userId Within Custom Tool/Function?

### DISCUSSION THREAD

**Discussion by dkindlund** [FlowiseAI/Flowise#2460] [Q&A]
@HenryHengZJ, is there any way to access the current $userId within a custom tool/function?

Ultimately, I'm trying to "pass through" that $userId into a downstream chatflow for proper Langfuse reporting.

**Comment by HenryHengZJ** [comment]
where is the `$userId` coming from?

**Reply by dkindlund** [reply]
Hey @HenryHengZJ , the `$userId` is originally provided to Flowise as a parameter in the ‘overrideConfig`, like:

```
output = query({
    "question": "Hey, how are you?",
    "overrideConfig": {
        "userId": "123"
    }
})
```

**Reply by HenryHengZJ** [reply]
not possible yet, so am I right to say you're looking for something like `$flow.overrideConfig.userId` on top of the rest of the available flow configurations - https://docs.flowiseai.com/integrations/langchain/tools/custom-tool#additional

**Reply by dkindlund** [reply]
Hey @HenryHengZJ , yes - exactly!  Having all of the `overrideConfig` data available via like a `$flow.overrideConfig` reference would be greatly appreciated!
